vendor/assets/javascripts/github_buttons.js

```diff
@@ -12,177 +12,89 @@
-  var button = document.querySelector('.gh-btn');
-  var mainButton = document.querySelector('.github-btn');
-  var text = document.querySelector('.gh-text');
-  var counter = document.querySelector('.gh-count');
-
-  // Constants
-  var LABEL_SUFFIX = ' on GitHub';
-  var GITHUB_URL = 'https://github.com/';
-  var API_URL = 'https://api.github.com/';
-  var REPO_URL = GITHUB_URL + user + '/' + repo;
-  var USER_REPO = user + '/' + repo;
-
-  window.callback = function(obj) {
-    if (obj.data.message === 'Not Found') {
-      return;
-    }
-
-    switch (type) {
-      case 'watch':
-        if (v === '2') {
-          counter.textContent = obj.data.subscribers_count && addCommas(obj.data.subscribers_count);
-          counter.setAttribute('aria-label', counter.textContent + ' watchers' + LABEL_SUFFIX);
+    function jsonp(path) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', path, true);
+      xhr.responseType = 'json';
+      xhr.onload = function() {
+        var status = xhr.status;
+        if (status === 200) {
+          var res = { data: { stargazers_count: xhr.response.stargazers_count } };
+          callback(res);
         } else {
-          counter.textContent = obj.data.stargazers_count && addCommas(obj.data.stargazers_count);
-          counter.setAttribute('aria-label', counter.textContent + ' stargazers' + LABEL_SUFFIX);
+          console.log("Request to github failed with status:", status)
         }
-
-        break;
-      case 'star':
-        counter.textContent = obj.data.stargazers_count && addCommas(obj.data.stargazers_count);
-        counter.setAttribute('aria-label', counter.textContent + ' stargazers' + LABEL_SUFFIX);
-        break;
-      case 'fork':
-        counter.textContent = obj.data.network_count && addCommas(obj.data.network_count);
-        counter.setAttribute('aria-label', counter.textContent + ' forks' + LABEL_SUFFIX);
-        break;
-      case 'follow':
-        counter.textContent = obj.data.followers && addCommas(obj.data.followers);
-        counter.setAttribute('aria-label', counter.textContent + ' followers' + LABEL_SUFFIX);
-        break;
-    }
-
-    // Show the count if asked and if it's not empty
-    if (count === 'true' && counter.textContent !== '') {
-      counter.style.display = 'block';
-      counter.removeAttribute('aria-hidden');
+      };
+      xhr.send();
     }
-  };
 
-  // Set href to be URL for repo
-  button.href = REPO_URL;
+    // Elements
+    var button = document.querySelector('.gh-btn');
+    var mainButton = document.querySelector('.github-btn');
+    var text = document.querySelector('.gh-text');
+    var counter = document.querySelector('.gh-count');
+
+    // Parameters
+    var user = mainButton.dataset.user;
+    var repo = mainButton.dataset.repo;
+    var type = mainButton.dataset.type;
+    var count = mainButton.dataset.count;
+    var size = mainButton.dataset.size;
+
+    // Constants
+    var LABEL_SUFFIX = ' on GitHub';
```

Comment:
`var` allows for mutation, could we use `const` instead here?

Reply:
Rubygems.org solely uses `var` at the moment 😭  so out of convention, I'm ok to keep it as is.
I think we should go through the js files and convert vars into `let`s and `const`s some time.